src/systems/filecoin_vm/runtime/impl/runtime.go

```diff
@@ -191,6 +193,35 @@ func (rt *VMContext) _createActor(codeID actor.CodeID, address addr.Address) {
 	rt._rtAllocGas(gascost.ExecNewActor)
 }
 
+func (rt *VMContext) DeleteActor(address addr.Address) {
+	ok := false
+
+	// An actor may delete itself.
+	if rt._actorAddress.Equals(address) {
+		ok = true
+	}
+
+	// Special case: StoragePowerActor may delete a StorageMinerActor.
+	addrCodeID, found := rt.GetActorCodeID(address)
+	if found &&
+		rt._actorAddress.Equals(addr.StoragePowerActorAddr) &&
+		addrCodeID.Is_Builtin() && addrCodeID.As_Builtin() == actor.BuiltinActorID_StorageMiner {
+
+		ok = true
+	}
```

Comment:
This coupling is totally unnecessary. Just have the SPA send a DeleteSelf() message to the SMA.

Reply:
#806